app/uk/gov/hmrc/exports/repositories/SubmissionRepository.scala

```diff
@@ -46,22 +46,36 @@ class SubmissionRepository @Inject() (val mongoComponent: MongoComponent)(implic
     ) with RepositoryOps[Submission] {
 
   override def classTag: ClassTag[Submission] = implicitly[ClassTag[Submission]]
-  override val executionContext = ec
+  override val executionContext: ExecutionContext = ec
+
+  override val lastUpdate: Option[String] = Some("lastUpdate")
 
   def addAction(uuid: String, action: Action): Future[Option[Submission]] = {
     val filter = Json.obj("uuid" -> uuid)
     val update = Json.obj("$addToSet" -> Json.obj("actions" -> Json.toJson(action)))
+
     findOneAndUpdate(filter, update)
   }
 
-  def updateAction(submissionId: String, actionId: String, decId: String): Future[Option[Submission]] =
+  def updateExternalAmendmentAction(uuid: String, actionId: String, declarationId: String): Future[Option[Submission]] = {
```

Comment:
This operation should possibly be reduced to a single `findOneAndUpdate`, in order to avoid conflicting concurrent updates to the document.

Reply:
Yes I agree should be one update. We shouldn't have a situation when the submission record is found but the action is missing so should be one atomic operation